api/handers_e2e_test.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand-sdk/encoding/json"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/indexer/api/generated/v2"
+	"github.com/algorand/indexer/idb"
+	"github.com/algorand/indexer/idb/postgres"
+	pgtest "github.com/algorand/indexer/idb/postgres/testing"
+	"github.com/algorand/indexer/util/test"
+)
+
+func setupIdb(t *testing.T, genesis bookkeeping.Genesis, genesisBlock bookkeeping.Block) (*postgres.IndexerDb /*db*/, func() /*shutdownFunc*/) {
+	_, connStr, shutdownFunc := pgtest.SetupPostgres(t)
+
+	db, _, err := postgres.OpenPostgres(connStr, idb.IndexerDbOptions{}, nil)
+	require.NoError(t, err)
+
+	err = db.LoadGenesis(genesis)
+	require.NoError(t, err)
+
+	err = db.AddBlock(&genesisBlock)
+	require.NoError(t, err)
+
+	return db, shutdownFunc
+}
+
+func TestApplicationHander(t *testing.T) {
+	db, shutdownFunc := setupIdb(t, test.MakeGenesis(), test.MakeGenesisBlock())
+	defer shutdownFunc()
+
+	///////////
+	// Given // A block containing an app call txn with ExtraProgramPages
+	///////////
+
+	const extraPages = 1
+	const expectedAppIdx = 1
+	txn := transactions.SignedTxnWithAD{
+		SignedTxn: transactions.SignedTxn{
+			Txn: transactions.Transaction{
+				Type: "appl",
+				Header: transactions.Header{
+					Sender:      test.AccountA,
+					GenesisHash: test.GenesisHash,
+				},
+				ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+					ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ExtraProgramPages: extraPages,
+				},
+			},
+			Sig: test.Signature,
+		},
```

Comment:
Technically, the ApplyData / ApplicationID should be set here. I think it's even used with the new accounting code.

Reply:
I tried and it did not work. I thought because the it uses re-evaluated txn and not txn from a block, but looks like it is [not the case](https://github.com/algorand/indexer/blob/develop/idb/postgres/internal/writer/writer.go#L202). I'll take another look.